src/foam/u2/view/GroupByView.js

```diff
@@ -0,0 +1,246 @@
+      //NO-OP
+    },
+    function getColumns(tc){
+      var data = this.data;
+      var arr = [];
+      var tableColumns = tc;
+      var topLevelProps = [];
+      for ( var i = 0 ; i < tableColumns.length ; i++ ) {
+        var rootProperty;
+        if ( foam.String.isInstance(tableColumns[i]) ) {
+          var axiom;
+          if ( tableColumns[i].includes('.') )
+            axiom = data.of.getAxiomByName(tableColumns[i].split('.')[0]);
+          else {
+            axiom = tableColumns.find(c => c.name === tableColumns[i]);
+            if ( ! axiom )
+              axiom = data.of.getAxiomByName(tableColumns[i]);
+          }
+          if ( ! axiom ) {
+            continue;
+          }
+          rootProperty = [ axiom.name, this.columnHandler.returnAxiomHeader(axiom) ];
+        } else {
+          rootProperty = tableColumns[i];
+          }
+        var rootPropertyName = this.columnHandler.checkIfArrayAndReturnPropertyNamesForColumn(rootProperty);
+        if ( ! topLevelProps.includes(rootPropertyName) ) {
+          arr.push(foam.u2.view.SubColumnGroupByConfig.create({
+            index:i,
+            rootProperty:rootProperty,
+            isPropertySelected: true,
+            level:0,
+            of:data.of,
+            selectedColumns$: data.selectedColumnNames$,
+          }, this));
+          topLevelProps.push(rootPropertyName);
+        }
+      }
+      arr.sort((a, b) => {
+        var aName = this.columnHandler.checkIfArrayAndReturnRootPropertyHeader(a.rootProperty);
+        var bName = this.columnHandler.checkIfArrayAndReturnRootPropertyHeader(b.rootProperty);
+        return aName.toLowerCase().localeCompare(bName.toLowerCase());
+      });
+      return arr;
+    }
+  ],
+  actions: [
+    {
+      name: 'resetColumns',
+      label: 'Reset Columns',
+      code: function() {
+        if ( this.selectCol ) {
+        this.data.groupBy = undefined;
+        this.selectCol.isPropertySelected = true;
+        this.selectCol = undefined;
+        }
+      }
+    }
+  ]
+  });
+  foam.CLASS({
+    package: 'foam.u2.view',
+    name: 'GroupByViewHeader',
+    extends: 'foam.u2.view.ColumnViewHeader',
+    imports: ['theme'],
+    requires: ['foam.u2.tag.Image'],
+    methods: [
+      function render() { 
+        var self = this;
+        this
+          .start()
+          .on('click', this.toggleSelection)
+            .addClass(this.myClass('some-padding'))
+            .style({
+              'padding-left': self.data.level * 16 + 8 + 'px',
+              'padding-right': '8px'
+            })
+            .start()
+              .addClass(this.myClass('label'))
+              .callIfElse( this.theme , 
+                function() {
+                  this.add(self.slot(function( data$isPropertySelected ) { 
+                    var image;
+                    if ( ! this.data.isPropertySelected ) {
+                      image = self.theme.glyphs.folderFill.getDataUrl({ fill: self.theme.primary3});
```

Comment:
```suggestion
                      image = self.theme.glyphs.folderFill.getDataUrl({ fill: self.theme.primary3 });
```
Same for the next one